### ISSUE TITLE
Error if scaling_max_freq unreadable

### ISSUE THREAD

**Issue by ocaisa** [hpcugent/easybuild-framework#2036]
`eb` gets a traceback if `/sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq` is not readable, probably just noting that it is unreadable is enough.

This would solve the problems I have running `eb` on [Bash on Ubuntu on Windows](https://msdn.microsoft.com/en-us/commandline/wsl/about) which I know is probably a big concern for you

**Comment by boegel** [comment]
Hmm, is this the only 'file' we need to be careful with w.r.t. read permissions, or are there others?

Have you run the framework unit tests in the Bash@Windows environment?

Primarily these: `python -O -m test.framework.systemtools`

**Comment by ocaisa** [comment]
Yes, just ran those tests and changing the read permissions allowed them to pass:
```
ocaisa@DESKTOP-1653BCN:~$ python -O -m test.framework.systemtools
INFO: This is (based on) vsc.install.shared_setup 0.10.20
.............E......................E
======================================================================
ERROR: test_cpu_speed_native (__main__.SystemToolsTest)
Test getting CPU speed.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ocaisa/EasyBuild/software/EasyBuild/3.0.1/lib/python2.7/site-packages/easybuild_framework-3.0.1-py2.7.egg/test/framework/systemtools.py", line 403, in test_cpu_speed_native
    cpu_speed = get_cpu_speed()
  File "/home/ocaisa/EasyBuild/software/EasyBuild/3.0.1/lib/python2.7/site-packages/easybuild_framework-3.0.1-py2.7.egg/easybuild/tools/systemtools.py", line 356, in get_cpu_speed
    txt = read_file(MAX_FREQ_FP)
  File "/home/ocaisa/EasyBuild/software/EasyBuild/3.0.1/lib/python2.7/site-packages/easybuild_framework-3.0.1-py2.7.egg/easybuild/tools/filetools.py", line 143, in read_file
    raise EasyBuildError("Failed to read %s: %s", path, err)
EasyBuildError: "Failed to read /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq: [Errno 13] Permission denied: '/sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq'"

======================================================================
ERROR: test_system_info (__main__.SystemToolsTest)
Test getting system info.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ocaisa/EasyBuild/software/EasyBuild/3.0.1/lib/python2.7/site-packages/easybuild_framework-3.0.1-py2.7.egg/test/framework/systemtools.py", line 655, in test_system_info
    system_info = get_system_info()
  File "/home/ocaisa/EasyBuild/software/EasyBuild/3.0.1/lib/python2.7/site-packages/easybuild_framework-3.0.1-py2.7.egg/easybuild/tools/systemtools.py", line 609, in get_system_info
    'cpu_speed': get_cpu_speed(),
  File "/home/ocaisa/EasyBuild/software/EasyBuild/3.0.1/lib/python2.7/site-packages/easybuild_framework-3.0.1-py2.7.egg/easybuild/tools/systemtools.py", line 356, in get_cpu_speed
    txt = read_file(MAX_FREQ_FP)
  File "/home/ocaisa/EasyBuild/software/EasyBuild/3.0.1/lib/python2.7/site-packages/easybuild_framework-3.0.1-py2.7.egg/easybuild/tools/filetools.py", line 143, in read_file
    raise EasyBuildError("Failed to read %s: %s", path, err)
EasyBuildError: "Failed to read /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq: [Errno 13] Permission denied: '/sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq'"

----------------------------------------------------------------------
Ran 37 tests in 25.496s

FAILED (errors=2)
ocaisa@DESKTOP-1653BCN:~$ sudo chmod a+r /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq
[sudo] password for ocaisa:
ocaisa@DESKTOP-1653BCN:~$ python -O -m test.framework.systemtools
INFO: This is (based on) vsc.install.shared_setup 0.10.20
.....................................
----------------------------------------------------------------------
Ran 37 tests in 23.349s

OK
ocaisa@DESKTOP-1653BCN:~$
```

**Comment by ocaisa** [comment]
This doesn't mean that everything works, I successfully installed `foss/2016b`  but failed to build `HPL` because the make seems to mistake a soft link for a directory on the system. I should open a ticket with them for that I suppose

**Comment by ocaisa** [comment]
I opened the ticket, the real problem is actually more basic:
https://github.com/Microsoft/BashOnWindows/issues/1503

**Comment by boegel** [comment]
check to make sure `scaling_max_freq` is readable before readying it is implemented in #2065, let's follow-up there
